Actualize current TopologyManager limits (#30806)
kubernetes/website: content/en/docs/tasks/administer-cluster/topology-manager.md

In the case of the `BestEffort` pod, the `static` CPU Manager policy would send back the default topology hint as there is no CPU request and the Device Manager would send back the hints for each of the requested devices.

Using this information the Topology Manager calculates the optimal hint for the pod and stores this information, which will be used by the Hint Providers when they are making their resource assignments.

### NUMA Topology-aware Scheduling

> The [out-of-tree scheduler](https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/noderesourcetopology) now could mitigate problem when kube-scheduler performs sub-optimal node assignments, without taking into account the behaviour of ToplogyManager on the worker node. To work properly this plugin should be subscribed to NodeResourceTopology modification generated by Node Resource Topology Exporter, whichis now part of [Node Feature Discovery] (https://github.com/kubernetes-sigs/node-feature-discovery).

Reviewer: There are a couple of typos here:

s/ToplogyManager /Topology Manager
s/whichis/which is

I think this can be phrased a bit better. We can move some the text in the know limitation section and use that to capture the motivation, something on the lines of:

```
The default scheduler is not topology-aware, so it is possible to be scheduled on a node and then fail on the node due to the Topology Manager. However, [out-of-tree scheduler](https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/noderesourcetopology) scheduler plugin mitigates this problem and allows optimal scheduling by taking resource topology into consideration.
In order to achieve this, the plugin needs to subscribe to NodeResourceTopology modification generated by Node Resource Topology Exporter which exposes resource information with NUMA node granularity.  An implementation of such an exporter is [Node Feature Discovery] (https://github.com/kubernetes-sigs/node-feature-discovery).
```

WDYT?

Author: ok, I agree to put it back to the "Known Limitations" as you suggested.

In the case of the `BestEffort` pod, the `static` CPU Manager policy would send back the default topology hint as there is no CPU request and the Device Manager would send back the hints for each of the requested devices.

Using this information the Topology Manager calculates the optimal hint for the pod and stores this information, which will be used by the Hint Providers when they are making their resource assignments.

### Known Limitations
1. The maximum number of NUMA nodes that Topology Manager allows is 8. With more than 8 NUMA nodes there will be a state explosion when trying to enumerate the possible NUMA affinities and generating their hints.
2. The default scheduler is not topology-aware, so it is possible to be scheduled on a node but then fail admission on the node due to the Topology Manager.
However, an [out-of-tree scheduler plugin](https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/noderesourcetopology) mitigates this problem and allows optimal scheduling by taking resource topology into consideration.

> In order to achieve this, the plugin needs to subscribe to a `NodeResourceTopology` Custom Resource endpoint.

Reviewer: This should be part of the 2nd point. This [renders](https://deploy-preview-30806--kubernetes-io-main-staging.netlify.app/docs/tasks/administer-cluster/topology-manager/) as a separate paragraph in the `Known Limitations` section and could possibly give the impression that we have moved to another topic.

In the case of the `BestEffort` pod, the `static` CPU Manager policy would send back the default topology hint as there is no CPU request and the Device Manager would send back the hints for each of the requested devices.

Using this information the Topology Manager calculates the optimal hint for the pod and stores this information, which will be used by the Hint Providers when they are making their resource assignments.

### Known Limitations
1. The maximum number of NUMA nodes that Topology Manager allows is 8. With more than 8 NUMA nodes there will be a state explosion when trying to enumerate the possible NUMA affinities and generating their hints.

> 2. The default scheduler is not topology-aware, it is possible to be scheduled on a node and then fail on the node due to the Topology Manager. However, an [out-of-tree scheduler plugin](https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/noderesourcetopology) mitigates this problem and allows optimal scheduling by taking resource topology into consideration.

Reviewer 2: ```suggestion
2. The default scheduler is not topology-aware, so it is possible to be scheduled on a node but then fail admission on the node due to the Topology Manager. However, an [out-of-tree scheduler plugin](https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/noderesourcetopology) mitigates this problem and allows optimal scheduling by taking resource topology into consideration.

```

> 2. The default scheduler is not topology-aware, it is possible to be scheduled on a node and then fail on the node due to the Topology Manager. However, [out-of-tree scheduler](https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/noderesourcetopology) scheduler plugin mitigates this problem and allows optimal scheduling by taking resource topology into consideration.

Reviewer 2: ```suggestion
2. The default scheduler is not topology-aware, it is possible to be scheduled on a node and then fail on the node due to the Topology Manager. However, an [out-of-tree scheduler plugin](https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/noderesourcetopology) mitigates this problem and allows optimal scheduling by taking resource topology into consideration.
```

> In order to achieve this, the plugin needs to subscribe to NodeResourceTopology modification generated by Node Resource Topology Exporter which exposes resource information with NUMA node granularity. [Node Feature Discovery](https://github.com/kubernetes-sigs/node-feature-discovery) implements Node Resource Topology Exporter in one of its plugins.

Reviewer 2: Can you clarify what a "`NodeResourceTopology` modification" is? Specifically around. the word "modification" -- what is meant by that?

Reviewer: "NodeResourceTopology modification" here refers to the  NodeResourceTopology CRs which capture per numa resource information corresponding to the nodes in the cluster. 

I think we can do a better job at phrasing this, maybe we change it something like below:

In order to achieve this, the plugin needs to access NodeResourceTopology CRs generated by Node Resource Topology Exporter which exposes resource information with NUMA node.....

> In order to achieve this, the plugin needs to subscribe to NodeResourceTopology modification generated by Node Resource Topology Exporter which exposes resource information with NUMA node granularity. [Node Feature Discovery](https://github.com/kubernetes-sigs/node-feature-discovery) implements Node Resource Topology Exporter in one of its plugins.

Reviewer 2: ```suggestion
In order to achieve this, the plugin needs to subscribe to a `NodeResourceTopology` modification generated by a Node Resource Topology Exporter which exposes NUMA information about the various resources available on the machine. [Node Feature Discovery](https://github.com/kubernetes-sigs/node-feature-discovery) implements the Node Resource Topology Exporter in one of its plugins.
```

> In order to achieve this, the plugin needs to subscribe to a `NodeResourceTopology` Custom Resource endpoint, the Custom Resources are generated by a Node Resource Topology Exporter which exposes NUMA information about the various resources available on the machine. [Node Feature Discovery](https://github.com/kubernetes-sigs/node-feature-discovery) implements the Node Resource Topology Exporter in one of its plugins.

Reviewer 2: ```suggestion
In order to achieve this, the plugin needs to subscribe to a `NodeResourceTopology` Custom Resource endpoint. These Custom Resources are generated by a Node Resource Topology Exporter which exposes NUMA information about the various resources available on the machine. [Node Feature Discovery](https://github.com/kubernetes-sigs/node-feature-discovery) implements the Node Resource Topology Exporter in one of its plugins.

```
